Enhance parameter filtering in get_optional_params_image_gen to prevent unsupported parameters from being added to extra_body for Azure GPT image generation

Dashing-Nelson opened the pull request:
## Title

<!-- e.g. "Implement user authentication feature" -->

This pull request improves parameter handling for Azure GPT image generation models to prevent unsupported parameters from being included in API requests, which previously could cause errors. It also adds a new test to ensure this behavior is maintained.

**Parameter filtering improvements:**

* Updated `_check_valid_arg` in `litellm/utils.py` to filter out unsupported parameters from `passed_params` when `drop_params` is enabled, ensuring only supported parameters are included in API requests. This prevents issues like sending `response_format` to Azure's `gpt-image-1` model, which can result in 400 Bad Request errors.

**Testing enhancements:**

* Added `test_azure_gpt_image_1_no_extra_body_injection` in `tests/test_litellm/test_utils.py` to verify that unsupported parameters (such as `response_format`) are dropped and not injected into `extra_body` when using provider config with Azure GPT image generation, and that supported parameters are preserved.

## Relevant issues

<!-- e.g. "Fixes #000" -->

Fixes #15028 

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [X] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [X] I have added a screenshot of my new test passing locally 
- [X] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [X] My PR's scope is as isolated as possible, it only solves 1 specific problem


## Type

<!-- Select the type of Pull Request -->
<!-- Keep only the necessary ones -->

🐛 Bug Fix

## Changes

Azure OpenAI Reject Image Generation Payload Due to extra_body Injection in LiteLLM v1.77.3 for gpt-image-1

Dashing-Nelson committed:
Enhance parameter filtering in get_optional_params_image_gen to prevent unsupported parameters from being added to extra_body for Azure GPT image generation

krrishdholakia commented:
hi @Dashing-Nelson, thank you for this PR! i think we already have a reusable function for this logic - why not use that? 

Also - your currently failing linting

Dashing-Nelson committed:
Refactor get_optional_params_image_gen to track and combine dropped parameters for improved handling of unsupported arguments

Dashing-Nelson committed:
Refactor get_optional_params_image_gen to improve parameter handling for image generation by introducing helper functions for provider-specific logic

Dashing-Nelson committed:
Refactor get_optional_params_image_gen to eliminate unused dropped_params variable and streamline argument validation logic

Dashing-Nelson committed:
Refactor get_optional_params_image_gen to track and combine dropped parameters for improved handling of unsupported arguments

Dashing-Nelson committed:
Refactor get_optional_params_image_gen to remove _process_kwargs_for_image_gen function, streamlining parameter processing for image generation

Dashing-Nelson committed:
Refactor get_optional_params_image_gen to introduce _process_kwargs_for_image_gen function, enhancing filtering of provider-specific parameters for image generation

Dashing-Nelson commented:
> hi @Dashing-Nelson, thank you for this PR! I think we already have a reusable function for this logic - why not use that?
> 
> Also - your currently failing linting - if you can look into that, that'd be great!

@krrishdholakia I have made use of the function and modified `_check_valid_arg`. I had missed it. It is now implemented.

I also noticed that linting and the test cases are failing for functions that are not related to mine.

Dashing-Nelson committed:
Refactor get_optional_params_image_gen to combine additional_drop_params with dropped_params for improved handling of unsupported parameters

Dashing-Nelson committed:
Merge branch 'BerriAI:main' into main

Dashing-Nelson committed:
fix(utils): ensure aspectRatio is an integer in _handle_vertex_ai_image_params

Dashing-Nelson committed:
Merge branch 'main' into main

Dashing-Nelson commented:
@plageat @krrishdholakia  any update?
